specifyweb/workbench/upload/treerecord.py

```diff
@@ -293,7 +295,7 @@ def to_json(self) -> Dict:
             rank_key = rank.rank_name if isinstance(rank, TreeRankRecord) else rank
             treeNodeCols = {k: v.to_json() if hasattr(v, "to_json") else v for k, v in cols.items()}
             
-            if len(cols) == 1:
+            if len(cols) == 1 and not isinstance(rank, TreeRankRecord):
                 result["ranks"][rank_key] = treeNodeCols["name"]
```

Comment:
Looks like this change breaks the test for unparsing an upload plan
https://github.com/specify/specify7/blob/7ff9c53f839046c4d136891258da94bbccd2b723/specifyweb/workbench/upload/tests/testschema.py#L37-L39

The expected value for the unparsed plan expects a TreeRankRecord for every rank. Could simply fix this by changing the original upload plan to include TreeRankRecords, which shouldn't break anything functionally as upload should work fine either way. Need to investigate if there are side effects

Reply:
Here's a diffcheck: https://www.diffchecker.com/7wYAWTtk/

Reply:
Removed this to fix the test rather than changing the test. Functionally, nothing should change since upload plan still works when the key name is in the format `<treename>~><rankname>`

Reply:
Looks like we need this to make the data mapper work correctly in #6126